frontend/templates/views/partials/survey.j2

```diff
@@ -1,15 +1,18 @@
 {% if survey_questions %}
+  [% set surveys = {{ survey_questions|tojson }} %]
+  [% set fixedQuestions = {{fixed_questions|tojson }} %]
+  [% set url = requestPath %]
+
   {% do doc.styles.addCssFile('css/components/organisms/survey.css') %}
 
   <amp-state id="surveyQuestions">
     <script type="application/json">
-
-      {% do survey_questions.update({"_url": request.url })%}
-      {{ survey_questions | tojson }}
+        [% set survey = surveys|random|survey(fixedQuestions, url) %]
+        [= survey|safe =]
     </script>
   </amp-state>
 
-  <amp-script src="/static/survey.js" width="450" height="1205"  id="fez-container" sandbox="allow-forms">
+  <amp-script src="<% hosts.platform %>/static/survey.js" width="450" height="1205"  id="fez-container" sandbox="allow-forms">
     <form id="fez">
```

Comment:
Can you also fix the form specific validation errors?

```
The mandatory attribute 'action' is missing in tag 'form'. 
The mandatory attribute 'target' is missing in tag 'form'. 
```

Reply:
pushed.